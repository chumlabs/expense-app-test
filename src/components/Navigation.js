// IMPORTS
// - libraries
import React, { Component } from 'react';
import { Link, NavLink } from 'react-router-dom';

const Navigation = () => (
  <nav>
    <ul>
      <li>
        <NavLink exact to="/" activeClassName="is-active">
          Home
        </NavLink>
      </li>
      <li>
        <NavLink to="/create" activeClassName="is-active">
          Create
        </NavLink>
      </li>
      <li>
        <NavLink to="/help" activeClassName="is-active">
          Help
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navigation;
