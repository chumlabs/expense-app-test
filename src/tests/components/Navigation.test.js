import React from 'react';
import { shallow } from 'enzyme';
import Navigation from '../../components/Navigation';

test('should render the Navigation component correctly', () => {
  const wrapper = shallow(<Navigation />);
  expect(wrapper).toMatchSnapshot();
});
