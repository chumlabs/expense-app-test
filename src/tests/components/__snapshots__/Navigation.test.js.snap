// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Navigation component correctly 1`] = `
<nav>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        to="/create"
      >
        Create
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        to="/help"
      >
        Help
      </NavLink>
    </li>
  </ul>
</nav>
`;
